odoo/tools/misc.py

```diff
@@ -1156,6 +1160,37 @@ def formatLang(env, value, digits=None, grouping=True, monetary=False, dp=False,
             res = '%s %s' % (currency_obj.symbol, res)
     return res
 
+def formatLangDate(env, value, lang_code=False, date_format=False):
+    '''
+        Formats the date in a given format.
+
+        :param env: an environment.
+        :param value: date, datetime or string. The date to format.
+        :param lang_code: string. The lang code, if not specified it is extracted from the
+            environment context.
+        :param date_format: string. The format or the date, if not specified the default format of
+            the lang.
+        :return: date formatted in the specified format.
+        :rtype: string
+    '''
+    if not value:
+        return ''
+    if isinstance(value, datetime.datetime):
+        value = value.date()
+    elif isinstance(value, basestring):
+        if len(value) < DATE_LENGTH:
+            return ''
+        else:
+            value = value[:DATE_LENGTH]
+            value = datetime.datetime.strptime(value, DEFAULT_SERVER_DATE_FORMAT).date()
```

Comment:
Possibly not, but in that case we probably want the entire behaviour in tools called from fields.